### PR TITLE
Remove EC2 metrics for serverless deployment

### PR DESCRIPTION
**Pull request by vmercierfr**
# Objective

Remove EC2 metrics for serverless deployment

# Why

AWS Aurora can be deployed as `serverless` deployment. Which make the EC2 metrics not relevant and report issues:

```
{"time":"2024-03-21T12:13:40.256259749Z","level":"ERROR","msg":"can't fetch EC2 metrics: can't fetch describe instance types: operation error EC2: DescribeInstanceTypes, https response error StatusCode: 400, RequestID: b0af5ca9-11a5-47da-b64b-d8187a399999, api error InvalidInstanceType: The following supplied instance types do not exist: [serverless]"}
```

While we are not (yet) supporting Aurora, we should avoid reaching EC2 metrics for serverless deployments.

# How

- Exclude `serverless` EC2 metrics

# Release plan

- [ ] Merge this PR
- [ ] Apply with CI

### CODE REVIEW

**Comment by vmercierfr** [comment]
@dcupif I updated the changes with your feedback, can you review it?

File: internal/app/exporter/ec2.go [hunk 1]
```diff
@@ -24,14 +22,11 @@ func getUniqTypeAndIdentifiers(instances map[string]rds.RdsInstanceMetrics) ([]s
 		}
 	}
 
-	return instanceIdentifiers, instanceTypes
-}
+	// Remove incompatible instance types
+	instanceTypes = RemoveElementsByValue(instanceTypes, []string{"db.serverless"})
```
Comment (dcupif) [inline internal/app/exporter/ec2.go:26]:
```suggestion
	supportedInstanceTypes := RemoveElementsByValue(instanceTypes, []string{"db.serverless"})
```

File: internal/app/exporter/ec2.go [hunk 2]
```diff
@@ -1,13 +1,11 @@
...
 
 import (
-	"regexp"
-
 	"github.com/qonto/prometheus-rds-exporter/internal/app/rds"
 	"golang.org/x/exp/slices"
 )
 
-func getUniqTypeAndIdentifiers(instances map[string]rds.RdsInstanceMetrics) ([]string, []string) {
+func GetUniqTypeAndIdentifiers(instances map[string]rds.RdsInstanceMetrics) ([]string, []string) {
```
Comment (dcupif) [inline internal/app/exporter/ec2.go:8]:
```suggestion
func getUniqTypeAndIdentifiers(instances map[string]rds.RdsInstanceMetrics) ([]string, []string) {
```
Seems like it does not need to be exported.

File: internal/app/exporter/prometheus.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package exporter
+
+import (
+	"regexp"
+)
+
+func ClearPrometheusLabel(str string) string {
```
Comment (dcupif) [inline internal/app/exporter/prometheus.go:7]:
```suggestion
func clearPrometheusLabel(str string) string {
```
Only used in this package, I would not export it.

File: internal/app/exporter/string.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package exporter
+
+import "golang.org/x/exp/slices"
+
+func RemoveElementsByValue(originalList, stringsToRemove []string) []string {
```
Comment (dcupif) [inline internal/app/exporter/string.go:5]:
```suggestion
func removeElementsByValue(originalList, stringsToRemove []string) []string {
```
Same here.

**Review by dcupif — APPROVED** [review]
LGTM with a last few adjustments 🚀

File: internal/app/exporter/ec2.go (hunk 1 above)
Comment (vmercierfr) [reply internal/app/exporter/ec2.go]: I applied the change

File: internal/app/exporter/ec2.go (hunk 2 above)
Comment (vmercierfr) [reply internal/app/exporter/ec2.go]: As discussed, I moved them to private function.

Helper:

> Go allows to place tests in a separate package with suffix _test. For example, tests for store package can be in the same package or in the package store_test. In the second case, you have to import the source code into tests so only exported things are available.
> The linter reports if a test is in a package without suffix _test. If you really need to test unexported function, then put the test into file XXX_internal_test.go. The linter skips such files by default. It also skips the file export_test.go by default (see the last article below).
> https://github.com/maratori/testpackage

File: internal/app/exporter/prometheus.go (hunk 3 above)
Comment (vmercierfr) [reply internal/app/exporter/prometheus.go]: Done, see https://github.com/qonto/prometheus-rds-exporter/pull/151#discussion_r1537362418

File: internal/app/exporter/string.go (hunk 4 above)
Comment (vmercierfr) [reply internal/app/exporter/string.go]: Done, see https://github.com/qonto/prometheus-rds-exporter/pull/151#discussion_r1537362418

**Pull request merged**